archive/demos/shiny-drum-machine.js

```diff
@@ -6,396 +10,171 @@ window.AudioContext = window.AudioContext || window.webkitAudioContext;
-    {"name":"Lopass Filter", "url":"impulse-responses/filter-lopass160.wav",    "dryMix":0, "wetMix":0.5},
-    {"name":"Hipass Filter", "url":"impulse-responses/filter-hipass5000.wav",   "dryMix":0, "wetMix":4.0},
-    {"name":"Comb 1", "url":"impulse-responses/comb-saw1.wav",                  "dryMix":0, "wetMix":0.7},
-    {"name":"Comb 2", "url":"impulse-responses/comb-saw2.wav",                  "dryMix":0, "wetMix":1.0},
-    {"name":"Cosmic Ping", "url":"impulse-responses/cosmic-ping-long.wav",      "dryMix":0, "wetMix":0.9},
-    {"name":"Kitchen", "url":"impulse-responses/house-impulses/kitchen-true-stereo.wav", "dryMix":1, "wetMix":1},
-    {"name":"Living Room", "url":"impulse-responses/house-impulses/dining-living-true-stereo.wav", "dryMix":1, "wetMix":1},
-    {"name":"Living-Bedroom", "url":"impulse-responses/house-impulses/living-bedroom-leveled.wav", "dryMix":1, "wetMix":1},
-    {"name":"Dining-Far-Kitchen", "url":"impulse-responses/house-impulses/dining-far-kitchen.wav", "dryMix":1, "wetMix":1},
-    {"name":"Medium Hall 1", "url":"impulse-responses/matrix-reverb2.wav",      "dryMix":1, "wetMix":1},
-    {"name":"Medium Hall 2", "url":"impulse-responses/matrix-reverb3.wav",      "dryMix":1, "wetMix":1},
-    {"name":"Large Hall", "url":"impulse-responses/spatialized4.wav",           "dryMix":1, "wetMix":0.5},
-    {"name":"Peculiar", "url":"impulse-responses/peculiar-backwards.wav",       "dryMix":1, "wetMix":1},
-    {"name":"Backslap", "url":"impulse-responses/backslap1.wav",                "dryMix":1, "wetMix":1},
-    {"name":"Warehouse", "url":"impulse-responses/tim-warehouse/cardiod-rear-35-10/cardiod-rear-levelled.wav", "dryMix":1, "wetMix":1},
-    {"name":"Diffusor", "url":"impulse-responses/diffusor3.wav",                "dryMix":1, "wetMix":1},
-    {"name":"Binaural Hall", "url":"impulse-responses/bin_dfeq/s2_r4_bd.wav",   "dryMix":1, "wetMix":0.5},
-    {"name":"Huge", "url":"impulse-responses/matrix-reverb6.wav",               "dryMix":1, "wetMix":0.7},
-]
-
-var impulseResponseList = 0;
-
-function ImpulseResponse(url, index) {
-    this.url = url;
-    this.index = index;
-    this.startedLoading = false;
-    this.isLoaded_ = false;
-    this.buffer = 0;
-    
-    this.demoIndex = -1; // no demo
-}
 
-ImpulseResponse.prototype.setDemoIndex = function(index) {
-    this.demoIndex = index;
-}
+class ImpulseResponse {
+    constructor(data, index) {
+        this.name = data.name;
+        this.url = data.url;
+        this.index = index;
+        this.buffer = undefined;
+    }
 
-ImpulseResponse.prototype.isLoaded = function() {
-    return this.isLoaded_;
-}
+    async load() {
+        if (!this.url) {
+            return; // "No effect" instance has empty URL.
+        }
 
-ImpulseResponse.prototype.load = function() {
-    if (this.startedLoading) {
-        return;
+        const request = new Request(this.url);
```

Comment:
Looks like this is the same code with loadSample() above.

It's fine as-is, but perhaps we can think about building a helper library for these basic functionality.

Reply:
Good idea, will do in follow up PR.